amqp_serde/src/types.rs

```diff
@@ -276,6 +277,30 @@ pub enum FieldValue {
     x(ByteArray), // RabbitMQ only
 }
 
+impl FieldValue {
+    pub fn len(&self) -> usize {
+        match self {
+            Self::t(_) => size_of::<Boolean>() + 1,
+            Self::b(_) => size_of::<ShortShortInt>() + 1,
+            Self::B(_) => size_of::<ShortShortUint>() + 1,
+            Self::s(_) => size_of::<ShortInt>() + 1,
+            Self::u(_) => size_of::<ShortUint>() + 1,
+            Self::I(_) => size_of::<LongInt>() + 1,
+            Self::i(_) => size_of::<LongUint>() + 1,
+            Self::l(_) => size_of::<LongLongInt>() + 1,
+            Self::f(_) => size_of::<Float>() + 1,
+            Self::d(_) => size_of::<Double>() + 1,
+            Self::D(_) => size_of::<DecimalValue>() + 1,
+            Self::S(l) => l.0 as usize + 5,
+            Self::A(f) => f.0 as usize + 5,
```

Comment:
How about to have associated constant, and make below code similar like below ?  
```suggestion
            Self::A(f) => f.0 as usize +  Self::TAG_SIZE + size_of_val(&l.0),
```

Reply:
Would you mind to add a test for testing `FieldTable`,?  I think it is not obvious if it works correctly.
please also check the `ci` failure.

Reply:
Still WIP, but I have already made some important changes